### PR TITLE
Fix contacts batches to one query request

### PR DESCRIPTION
**Pull request by barnahum8**
remove batches, added load more option.
checking for better result

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 1]
```diff
@@ -29,16 +29,11 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
...
-        currentPage,
-        setIsMore,
-        contactsLength, 
-        setContactsLength
+        getValues
     } = parameters;
 
     const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
     const datesToInvestigate = useSelector<StoreStateType, Date[]>(state => state.investigation.datesToInvestigate);
-    const isViewMode = useSelector<StoreStateType, boolean>(state => state.investigation.isViewMode);
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:41]: to add back

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 2]
```diff
@@ -78,7 +73,7 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
         const rulerLogger = logger.setup('client ruler logger setup');
         rulerLogger.info(`launching server request with parameter: ${JSON.stringify(RulerCheckColorRequestParameters)}`, Severity.LOW);
         setIsLoading(true);
-        return await axios.post('/ruler/rulerapi', RulerCheckColorRequestParameters,{timeout: 5000})
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:81]: to add back

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 3]
```diff
@@ -297,7 +295,8 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
                     Severity.LOW
                 );
             })
-    };
+            .finally(() => setIsLoading(false));
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:298]: to check if its the right place

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 4]
```diff
@@ -331,7 +330,7 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
         e.preventDefault();
         const data = getValues();
         const parsedFormData = parseFormBeforeSending(data as FormInputs);
-        if (!areThereDuplicateIds(data) || isViewMode) {
+        if (!areThereDuplicateIds(data)) {
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:333]: to add back

File: server/src/ClientToDBAPI/RulerRoute/mainRoutes.ts [hunk 5]
```diff
@@ -28,8 +28,8 @@ rulerRoute.post('/rulerapi/', (req: Request, res: Response) => {
         }
         ).catch((err: any) => {
             rulerLogger.error(invalidAPIResponseLog(err), Severity.HIGH);
-            res.status(errorStatusCode).send(err);
```
Comment (chenFriedman) [inline server/src/ClientToDBAPI/RulerRoute/mainRoutes.ts:31]: add back

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts (hunk 4 above)
Comment (barnahum8) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts]: 👍

File: server/src/ClientToDBAPI/RulerRoute/mainRoutes.ts (hunk 5 above)
Comment (barnahum8) [reply server/src/ClientToDBAPI/RulerRoute/mainRoutes.ts:31]: 👍

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts (hunk 3 above)
Comment (barnahum8) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts]: 👍

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts (hunk 2 above)
Comment (barnahum8) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:81]: 👍

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts (hunk 1 above)
Comment (barnahum8) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts]: 👍

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**